sofana/R3BSofFragmentAnalysis.cxx

```diff
@@ -267,11 +267,22 @@ void R3BSofFragmentAnalysis::Exec(Option_t* option)
     HitMwpc1 = new R3BMwpcHitData*[nHitMwpc1];
     HitMwpc2 = new R3BMwpcHitData*[nHitMwpc2];
     HitMwpc3 = new R3BMwpcHitData*[nHitMwpc3];
-
-    if (nHitMwpc0 < 1 || nHitMwpc1 < 1 || nHitMwpc2 < 1 || nHitMwpc3 < 1 || nHitTofW < 1 || nHitMusic < 1 ||
-        nHitTwim < 1)
+    //
+    if (nHitMusic < 1)
         return;
-
+    // Getting Music angle
+    for (Int_t ihit = 0; ihit < nHitMusic; ihit++)
+    {
+        HitMusic[ihit] = (R3BMusicHitData*)fMusicHitDataCA->At(ihit);
+        if (!HitMusic[ihit])
+            continue;
+        // In case the MusicHitData container has several "realistic" values,
+        // it's not possible to distinguish which is the "correct" event. Thus skipping events having several hits.
+        if (TMath::Abs(MusicTheta) < 0.1)
+            return;
+        MusicTheta = HitMusic[ihit]->GetTheta();
+    }
```

Comment:
I think that you should ask for this condition before starting with the next calculations:
if (nHitMwpc0 < 1 || nHitMwpc1 < 1)
        return;

Reply:
Thanks, I amended the code accordingly.